carousel-app/src/util/shuffle.js

```diff
@@ -0,0 +1,12 @@
+function shuffle(data){
+
+    //Fisher Yates Shuffle
+    
+    for(let i = data.length-1;i>=0;i--){
+        let j = Math.floor(Math.random() * (i + 1)); // random index from 0 to i
+        [data[i], data[j]] = [data[j],data[i]];
+    }
+    return data    
+}
```

Comment:
Can you explain why you decided to perform the shuffle on the client side?

Reply:
Yes, usually you perform sorting in the backend as you don't want your frontend to take time to load in order to show output to the user. But when the array size to sort is less than 20(in our case) , the performance difference between sorting in the backend vs sorting in the frontend is insignificant.  Plus as you see here the time complexity of this algorithm in worst case is O(n) as compared to a sort function which usually is O(nlogn) depending on the type of sort algorithm. On the whole this has to do with my implementation convenience and plus  that is why instead of implementing default sort I went with an algorithm that has better time complexity. Considering very large data size it is best to perform sort in the backend.